### PR TITLE
feat : #1

### LINKED ISSUE #1: user register api
- [x] server side validation for register inputs
- [x] save user to db
- [x] return new user data and authorization token

### PR DESCRIPTION
**Pull request by pasan-nadeeja-dev**
close #1

### CODE REVIEW

File: graphql/resolvers/users/register.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      $or: [{ username: username }, { email: email }],
+    },
+    { username: 1, email: 1 }
+  );
+  if (existingUser) {
+    if (existingUser.username === username) {
+      errors.username = "Username is taken";
+    }
+    if (existingUser.email === email) {
+      errors.email = "Email is taken";
```
Comment (lead-manager) [inline graphql/resolvers/users/register.js:35]: Try refactor this block

**Review by lead-manager — CHANGES_REQUESTED** [review]
https://github.com/pasan-nadeeja-dev/SocialMeadiaApp/pull/2/files#r637571626

File: graphql/resolvers/users/register.js (hunk 1 above)
Comment (pasan-nadeeja-dev) [reply graphql/resolvers/users/register.js:35]: Done

**Review by lead-manager — APPROVED** [review]
LGTM

**Pull request merged**